core/main.go

```diff
@@ -4,27 +4,41 @@ import (
 	"fmt"
 	"log"
 	"os"
+	"strings"
 
 	"github.com/Masterminds/semver/v3"
 
 	"github.com/smartcontractkit/chainlink/core/cmd"
-	"github.com/smartcontractkit/chainlink/core/logger"
 	"github.com/smartcontractkit/chainlink/core/recovery"
 	"github.com/smartcontractkit/chainlink/core/static"
 )
 
 func init() {
 	// check version
 	if static.Version == static.Unset {
-		if os.Getenv("CHAINLINK_DEV") == "true" {
+		if isDevMode() {
 			return
 		}
-		log.Println(`Version was unset but CHAINLINK_DEV was not set to "true". Chainlink should be built with static.Version set to a valid semver for production builds.`)
+		log.Println(`Version was unset but dev mode is enabled. Chainlink should be built with static.Version set to a valid semver for production builds.`)
 	} else if _, err := semver.NewVersion(static.Version); err != nil {
 		panic(fmt.Sprintf("Version invalid: %q is not valid semver", static.Version))
 	}
 }
 
+// https://app.shortcut.com/chainlinklabs/story/33622/remove-legacy-config
+func isDevMode() bool {
```

Comment:
I know that it's slightly against [one of the mantras](https://www.youtube.com/watch?v=PAAkCSZUG1c&t=568s) but I feel like it might be easier down the line (when we inevitably remove `CHAINLINK_DEV`) to have this as a helper in the `config`/`envvar` package?

Reply:
I thought about it, but these are highly exceptional cases that will be removed during the next dev cycle, so I hesitated to create new API surface... :thinking:

Reply:
Yup I guess it should be fine, if you grep for the ticket ID (33622) you'd find all occurrences 😃

Reply:
Can you tell I'm excited to start deleting? :laughing: